src/Backends/Banshee.GStreamer/Banshee.GStreamer/PlayerEngine.cs

```diff
@@ -209,9 +209,11 @@ protected override void Initialize ()
 
         public override void Dispose ()
         {
+            bp_stop (handle, true);
```

Comment:
Yes. `gst_deinit` blocks until all GStreamer tasks are finished, so we need to stop playback before calling.